apps/condo/domains/meter/schema/MeterReadingExportTask.js

```diff
@@ -0,0 +1,250 @@
+    fields: {
+        status: {
+            schemaDoc: 'Status of the job. Used at client side progress information component',
+            type: 'Select',
+            options: EXPORT_STATUS_VALUES,
+            defaultValue: PROCESSING,
+            isRequired: true,
+            access: {
+                create: canOnlyServerSideWithoutUserRequest,
+                read: true,
+                update: true,
+            },
+        },
+
+        format: {
+            schemaDoc: 'Requested export file format',
+            type: 'Select',
+            options: [EXCEL],
+            isRequired: true,
+            access: {
+                create: true,
+                read: true,
+                update: canOnlyServerSideWithoutUserRequest,
+            },
+        },
+
+        exportedRecordsCount: {
+            schemaDoc: 'How many records at the moment are exported',
+            type: 'Integer',
+            isRequired: true,
+            defaultValue: 0,
+            access: {
+                create: canOnlyServerSideWithoutUserRequest,
+                read: true,
+                update: canOnlyServerSideWithoutUserRequest,
+            },
+        },
+
+        totalRecordsCount: {
+            schemaDoc: 'Total records to export that will be determined at server side in export operation',
+            type: 'Integer',
+            isRequired: true,
+            defaultValue: 0,
+            access: {
+                create: canOnlyServerSideWithoutUserRequest,
+                read: true,
+                update: canOnlyServerSideWithoutUserRequest,
+            },
+        },
+
+        file: {
+            schemaDoc: 'Meta information about file, saved outside of database somewhere. Shape of meta information JSON object is specific to file adapter, used by saving a file.',
+            type: 'File',
+            adapter: MeterReadingExportTaskFileAdapter,
+            access: {
+                create: canOnlyServerSideWithoutUserRequest,
+                read: true,
+                update: canOnlyServerSideWithoutUserRequest,
+            },
+        },
+
+        meta: {
+            schemaDoc: 'Stores information about query and ids of exported and failed records',
+            type: 'Json',
+            access: {
+                create: canOnlyServerSideWithoutUserRequest,
+                read: true,
+                update: canOnlyServerSideWithoutUserRequest,
+            },
+        },
+
+        where: {
+            schemaDoc: 'Filtering conditions for records to export',
+            type: 'Json',
+            isRequired: true,
+            access: {
+                read: true,
+                create: true,
+                update: false,
+            },
+            hooks: {
+                resolveInput: ({ resolvedData, fieldPath, operation }) => {
+                    if (operation !== 'create') return resolvedData[fieldPath]
```

Comment:
Is it necessary to add something like `graphQLReturnType: 'MeterReadingWhereInput'` here so that it is clear what JSON is expected?

Reply:
I don't think so. Just pass it like other Task schemas

Reply:
Why not just `MeterReadingWhereInput`, so API is type-friendly?